internal/replication/blockreplicator.go

```diff
@@ -719,6 +746,10 @@ func (br *BlockReplicator) setLastCommittedBlock(block *types.Block) {
 	defer br.mutex.Unlock()
 
 	br.lastCommittedBlock = block
+	if br.numInFlightBlocks > 0 { // only reduce on the leader
```

Comment:
this check is nothing but a `AmIALeader()` check?

Reply:
`isLeader && num>0`, for double safety, yes